docs/src/proposals/unified-scheduler.md

```diff
@@ -0,0 +1,96 @@
+# Unified Scheduler
+
+## Architechtural problem
+
+- there's two execution scheduling algorithms, and both are subotimmal for current tx load. 
+independent algorithms are exposing the so-called replayability risk
+
+also, increasing banking threads doesn't linearly scale to the number of cpu cores.
+ 
+## Current Transaction load pattern
+
+## Redefined scheduler's design goal
+
+saturate N cores with full of tx exec cycles.
+
+## Proposed Solution
+
+## why unify?: because simply using the same algo makes most sense.
+
+## Currnet implementation problem
+
+- why batching isn't good?
+- what about block max compute unit?
+
+## Proposed implementation
+
+### high-level scheduling algorithm
+
+eventual successful locking.
+
+pessimic locking is chosen for now (might switch to optimistic)
+
+
+### design philophy
+
+- no timer/no bucketing (thus, no jitter and no odd juggy incentive)
+- no huetrics (try to be generic/agnostic/adaptive as much as possible)
+- no estimator/prediction (so bots can't trick leaders)
+- single threaded
+- determinicity
+- strict fairness, only considering priority_fee/fcfs
+- approx. O(n) where n is gross total of addresses in transactions.
+- strict adherance to local fee market
```

Comment:
this seems more fair from the user perspective, but it may not yield the best fee-collection results for validators. A greedy validator will not adhere to this and we have no way to enforce it.

Just making a note, not saying it shouldn't be your goal; it may be the case that you can make some optimizations this way which *does* result in higher throughput and fee-collection!

Reply:
> this seems more fair from the user perspective, but it may not yield the best fee-collection results for validators. A greedy validator will not adhere to this and we have no way to enforce it.

yep. greedy validator will just seek to mev revenue, rather than fee-collection maximization, considering relatively cheap tx fees. cc: @buffalu 

> Just making a note, not saying it shouldn't be your goal; it may be the case that you can make some optimizations this way which does result in higher throughput and fee-collection!

yep, i'll write that reasoning down later for any scrutinization.